src/main/java/step3/model/Car.java

```diff
@@ -0,0 +1,21 @@
+package step3.model;
+
+import java.util.function.UnaryOperator;
+
+public class Car {
+    private static final UnaryOperator<String> validateName = name -> {
+        if (name.length() <= 5) {
+            return name;
+        }
+        throw new IllegalArgumentException("자동차 이름은 5자를 초과할수 없다");
+    };
```

Comment:
```suggestion
    private void validate(String name) {
        if (name.length() > 5) {
            throw new IllegalArgumentException("자동차 이름은 5자를 초과할수 없다");
        }
    }
```

평범한 void 함수 대신에 UnaryOperator 를 사용하신 의도가 있으시나요?
이건 정말 궁금해서 여쭈어봅니다.
꼭 장점이 아니더라도, Dong 은 어떨때 UnaryOperator 를 사용하시는지 궁금해서 여쭈어봅니다.

가령 저는 다음과 같은 기준으로 람다식을 사용합니다.
error 를 throw 하는 것처럼 side-effect 가 있는 로직이라면, 저는 전통적인 절차지향 스타일로 메서드를 선언하고,
람다식을 사용할 때에는 가능한 side-effect 를 스스로 최소화하는 방향으로 구현을 합니다.

하지만 이건 저만의 스타일과 저만의 코딩 스타일인지라,
Dong 은 람다식을 어떻게 활용하시는지 궁금해서 여쭈어봅니다.

Reply:
요거는 람다식에 큰 의미가 있다기 보다는 편의성때문에 쓰는데요 예를들어서.. 
`void validate(String name)`  이런식으로 함수를 쓰면 한줄을 무조건 더 써야하는데
`UnaryOperator<String> validateName` 으로 쓰면 validate 를 위해서 한줄을 더 추가하지 않아서 썻습니다..

Reply:
오호 그렇군요!
그런 의도라면 확실히 편리한것 같습니다!

Reply:
어제 제가 왜 UnaryOperator 가 문제될 것이 없는데 왜 의도에 대해 궁금해했을까 다시 고민을 해보았는데,
아무래도 제가 코틀린으로 개발하던 시절에 습관이 남아서 그랬던것 같습니다.
코틀린에서는 UnaryOperator 와 비슷하지만, 의미적으로 다르게 쓰는 [apply, with, let, also, run](https://medium.com/@limgyumin/%EC%BD%94%ED%8B%80%EB%A6%B0-%EC%9D%98-apply-with-let-also-run-%EC%9D%80-%EC%96%B8%EC%A0%9C-%EC%82%AC%EC%9A%A9%ED%95%98%EB%8A%94%EA%B0%80-4a517292df29) 가 있어서요.
가독성 측면에서 apply, with, let, also, run 을 영어 의미에 맞게 사용했는가를 코드리뷰 할 때 해주곤 했는데,
하지만 Java 에서는 오직 apply 만 한개 존재하니, 지금처럼 Dong 이 사용하신게 적절한것 같습니다.